Updated downloaded methods for ARIN files from ftp to https

sswallace opened the pull request:
ARIN has depricated its FTP servcies to download IRR files

sswallace committed:
Update __init__.py

Updating the download method for ARIN files from ftp to https

romain-fontugne commented:
There is a problem with a few GTT prefixes in ARIN's IRR that we cannot correctly parse (for example '065.084.000.000/15'). 
We are currently ignoring those. The new code should be available with `pip install rov`.

sswallace commented:
> There is a problem with a few GTT prefixes in ARIN's IRR that we cannot correctly parse (for example '065.084.000.000/15'). We are currently ignoring those. The new code should be available with `pip install rov`.

def normalize_ip(ip_str: str) -> str:
    """
    If ip_str is IPv6 (contains ':'), return it unchanged.
    If IPv4, remove leading zeros from each octet, preserving any /prefix.
    """
    s = ip_str.strip()
    # Quick IPv6 check
    if ':' in s:
        return s

    # Split off any network prefix
    if '/' in s:
        addr, prefix = s.split('/', 1)
    else:
        addr, prefix = s, None

    # Strip leading zeros in each octet
    octets = addr.split('.')
    stripped = [str(int(o)) for o in octets]  # int('007') → 7, str(7) → '7'
    normalized = '.'.join(stripped)

romain-fontugne commented:
Thanks! I will add that
